resource-aggregate/events/deviceMetadataSnapshotTaken.go

```diff
@@ -300,6 +300,9 @@ func (d *DeviceMetadataSnapshotTaken) updateDeviceConnection(ctx context.Context
 		return nil, status.Errorf(codes.InvalidArgument, "cannot update connection status for empty connectionId")
 	}
 	if !req.GetConnection().IsOnline() {
+		if em.GetVersion() == 0 {
```

Comment:
Version value starts from 1 if set? 

Otherwise we cannot distinguish been when `em` is `nil` and `em` is not `nil`, but `version` is equal to 0.